source/slang/slang-ir-autodiff.cpp

```diff
@@ -126,20 +126,22 @@ static IRInst* _getDiffTypeWitnessFromPairType(
 
 bool isNoDiffType(IRType* paramType)
 {
-    if (auto attrType = as<IRAttributedType>(paramType))
+    for(;;)
```

Comment:
I would just write while(parameter) here to get rid of any compiler warnings on unreachable code.